File: xmlplot.m
clear
close all
clc

% Load the original image
img = imread('x.jpg');

% Load the updated XML file
xml_file = xmlread('x.xml');

% Find all object elements in the XML file
object_nodes = xml_file.getElementsByTagName('object');

% Loop through all object elements and draw the bounding boxes on the image
for i = 0:object_nodes.getLength-1
    % Extract the coordinates from the XML file
    xmin = str2double(object_nodes.item(i).getElementsByTagName('xmin').item(0).getFirstChild.getData);
    ymin = str2double(object_nodes.item(i).getElementsByTagName('ymin').item(0).getFirstChild.getData);
    xmax = str2double(object_nodes.item(i).getElementsByTagName('xmax').item(0).getFirstChild.getData);
    ymax = str2double(object_nodes.item(i).getElementsByTagName('ymax').item(0).getFirstChild.getData);
    
    % Draw the bounding box on the image
    img = insertShape(img, 'Rectangle', [xmin, ymin, xmax-xmin, ymax-ymin], 'LineWidth', 3, 'Color', 'red');
end

% Display the image with the bounding boxes
imshow(img);

File: xmlplot.py
import matplotlib.pyplot as plt
import xml.etree.ElementTree as ET

# Load the image and its label in xml format
img = plt.imread('2.jpg')
tree = ET.parse('2.xml')
root = tree.getroot()

# Plot the image
fig, ax = plt.subplots()
ax.imshow(img)

# Loop through the objects and plot them on the image
for obj in root.iter('object'):
    label = obj.find('name').text # This should point to the 'name' tag inside each 'object' tag
    bndbox = obj.find('bndbox')
    xmin = int(round(float(bndbox.find('xmin').text)))
    ymin = int(round(float(bndbox.find('ymin').text)))
    xmax = int(round(float(bndbox.find('xmax').text)))
    ymax = int(round(float(bndbox.find('ymax').text)))
    rect = plt.Rectangle((xmin, ymin), xmax-xmin, ymax-ymin, fill=False, edgecolor='g', linewidth=2)
    ax.add_patch(rect)
    ax.text(xmin, ymin-5, label, color='g', fontsize=10)

# Show the image with labels
plt.show()
